Source/Dafny/Triggers/QuantifiersCollection.cs

```diff
@@ -299,32 +307,42 @@ internal class QuantifierGroup {
         }
 
         AddTriggersToMessage("Selected triggers:", q.Candidates, msg, indent);
+        if (q.Candidates.Count > 0) {
+          SetRelatedLocationMessage("Trigger position:");
+        }
         AddTriggersToMessage("Rejected triggers:", q.RejectedCandidates, msg, indent, true);
+        if (q.RejectedCandidates.Count > 0 && q.Candidates.Count == 0) {
+          SetRelatedLocationMessage("Rejected trigger position:");
+        }
 
 #if QUANTIFIER_WARNINGS
-        var WARN_TAG = DafnyOptions.O.UnicodeOutput ? "⚠ " : "/!\\ ";
+        var WARN_TAG = DafnyOptions.O.UnicodeOutput ? "⚠ " : @"/!\ ";
         var WARN_TAG_OVERRIDE = suppressWarnings ? "(Suppressed warning) " : WARN_TAG;
         var WARN_LEVEL = suppressWarnings ? ErrorLevel.Info : ErrorLevel.Warning;
         var WARN = indent + WARN_TAG_OVERRIDE;
         if (!q.CandidateTerms.Any()) {
           errorLevel = WARN_LEVEL;
           msg.Append(WARN).AppendLine("No terms found to trigger on.");
+          SetRelatedLocationMessage("Potential trigger not suitable:");
         } else if (!q.Candidates.Any()) {
           errorLevel = WARN_LEVEL;
           msg.Append(WARN).AppendLine("No trigger covering all quantified variables found.");
+          SetRelatedLocationMessage("Potential trigger not suitable:");
```

Comment:
Same here: what does this mean?  There are many trigger candidates within a term, so why highlight one?

Reply:
Because this trigger candidate is the only one that is being recorder in `q.tok` as a nested token. Otherwise, there would be no good reason to not include all of them, except for the legibility of the message. The alternative would be to get rid of this message. What do you think?

Reply:
Let's get rid of the message and of the nested token